### ISSUE TITLE
KeyError when using irbempy.get_Lstar

### ISSUE THREAD

**Issue by Rhyst223** [spacepy/spacepy#103]
My issue is ...

### Minimal example to reproduce issue:

I am trying to use the get_Lstar function in irbempy and am receiving the following error:

```
Traceback (most recent call last):

  File "<ipython-input-15-3f1adb4c9548>", line 1, in <module>
    Lstar(t,y)

  File "/usr/local/lib/python3.7/site-packages/spacepy/irbempy/irbempy.py", line 1566, in get_Lstar
    DALL = _get_Lstar(ticks, loci, alpha, extMag, options, omnivals)

  File "/usr/local/lib/python3.7/site-packages/spacepy/irbempy/irbempy.py", line 1296, in _get_Lstar
    d = prep_irbem(ticks, loci, alpha, extMag, options, omnivals)

  File "/usr/local/lib/python3.7/site-packages/spacepy/irbempy/irbempy.py", line 1614, in prep_irbem
    omnivals = omni.get_omni(ticks)

  File "/usr/local/lib/python3.7/site-packages/spacepy/omni.py", line 245, in get_omni
    enval, stval = omnirange(dbase=ldb)[1], omnirange(dbase=ldb)[0]

  File "/usr/local/lib/python3.7/site-packages/spacepy/omni.py", line 369, in omnirange
    start, end = hfile['RDT'][0], hfile['RDT'][-1]

  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper

  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper

  File "/Users/rlt1917/Library/Python/3.7/lib/python/site-packages/h5py/_hl/group.py", line 262, in __getitem__
    oid = h5o.open(self.id, self._e(name), lapl=self._lapl)

  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper

  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper

  File "h5py/h5o.pyx", line 190, in h5py.h5o.open

KeyError: "Unable to open object (object 'RDT' doesn't exist)"
```

### OS, Python version, and dependency version information:

I have every dependency installed except the CDF library. It looks like something is going on with h5py does anybody have any insight? 
I am using macOS Mojave and Python 3.7

**Comment by jtniehof** [comment]
You need the OMNI data for this to run, which are downloaded with toolbox.update(). That does, however, require the CDF library to process the downloaded data into the format that the spacepy omni module uses.

**Comment by Rhyst223** [comment]
Oh that makes sense, what is the easiest way to download the CDF library, the documentation on its website is quite confusing?

**Comment by jtniehof** [comment]
Probably easiest on Mac is to grab it from https://spdf.sci.gsfc.nasa.gov/pub/software/cdf/dist/cdf37_1/macosx/cdf37_1_0-setup_universal_binary.tar.gz

You don't need to reinstall SpacePy after installing the CDF library. Just check that "import spacepy.pycdf" works. pycdf checks any place on library paths, and also in /Applications/, /usr/local/, and the home directory.

I'm also working on the ability to have pycdf install the library for you, but that's a fairly big project.

**Comment by Rhyst223** [comment]
I have managed to install the CDF library successfully and it imports into Python okay, however, I am still receiving the same error as before. Might the CDF library have to be saved in a particular place?

**Comment by jtniehof** [comment]
Did you also run `toolbox.update`? If you `import omni` my guess is you'll get a message `Qin-Denton/OMNI2 data not found in current format.` That also gives the command to run to download the data (`spacepy.toolbox.update(QDomni=True)`)

These are the solar wind inputs necessary for the Lstar calculations.

**Comment by Rhyst223** [comment]
I've now attempted to update and get the following error:
```
spacepy.toolbox.update(all=True)
Retrieving Qin_Denton file ...
Traceback (most recent call last):

  File "<ipython-input-6-319cf0799ae4>", line 1, in <module>
    spacepy.toolbox.update(all=True)

  File "/usr/local/lib/python3.7/site-packages/spacepy/toolbox/__init__.py", line 756, in update
    u.urlretrieve(config['qindenton_url'], omni_fname_zip, reporthook=progressbar)

  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 247, in urlretrieve
    with contextlib.closing(urlopen(url, data)) as fp:

  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)

  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 525, in open
    response = self._open(req, data)

  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 543, in _open
    '_open', req)

  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 503, in _call_chain
    result = func(*args)

  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 1551, in ftp_open
    raise exc.with_traceback(sys.exc_info()[2])

  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 1533, in ftp_open
    fw = self.connect_ftp(user, passwd, host, port, dirs, req.timeout)

  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 1555, in connect_ftp
    persistent=False)

  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 2375, in __init__
    self.init()

  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 2387, in init
    self.ftp.cwd(_target)

  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ftplib.py", line 631, in cwd
    return self.voidcmd(cmd)

  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ftplib.py", line 278, in voidcmd
    return self.voidresp()

  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ftplib.py", line 251, in voidresp
    resp = self.getresp()

  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ftplib.py", line 246, in getresp
    raise error_perm(resp)

URLError: <urlopen error ftp error: error_perm('550 Failed to change directory.')>
```

**Comment by Rhyst223** [comment]
Okay I have managed to get around the problem. If I update to
``` pip3 install --upgrade spacepy==0.1.6```
and then run

`spacepy.toolbox.update(QDomni=True)`

it runs without fault. But then trying to access irbempy gives an error, not being able to locate the library. So if I revert back to spacepy 0.1.5 AFTER the toolbox update in 0.1.6 then the get_Lstar works fine.

So there is something happening in 0.1.6 which stops irbempy from working

**Comment by jtniehof** [comment]
My guess is this 0.1.6 problems might be connected to the problems you were having earlier with getting the Fortran compiler happy. You can try passing `--no-cache-dir` to pip but that might not affect things. pip isn't supposed to cache the build directory on recent versions but that's what this sounds like. We did update the irbem library itself between 0.1.5 and 0.1.6 so that might be an issue but it hasn't come up before.

**Comment by jtniehof** [comment]
Ah, `--no-cache-dir` also disabled the cache of built wheels, so may be worth a try. ([docs](https://pip.pypa.io/en/stable/reference/pip_install/#caching))

**Comment by Rhyst223** [comment]
I tried to update with `--no-cache-dir` but the issue with irbempy is still present. I think I'll stick with 0.1.5 now that the toolbox is updated, since everything seems to be working now, but perhaps this is an issue with irbempy in 0.1.6?

**Comment by drsteve** [comment]
The reason `update` doesn't work in 0.1.5 is, I believe, the URLs for data download changed. Quite why 0.1.6 wasn't working still isn't clear, but I'm not sure there's enough info here to nail down the issue.

I'm going to close this as a stale issue and it can be reopened if there's a need. Hopefully recent updates for numpy 1.16 (and gcc 8) compatibility also fix this for the upcoming 0.1.7 release.
